app/src/components/market/market_view_container.tsx

```diff
@@ -1,21 +1,65 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 
 import { useMarketMakerData } from '../../hooks'
 import { useConnectedWeb3Context } from '../../hooks/connectedWeb3'
 import { Loading } from '../common'
 
 import { MarketView } from './market_view'
+import gql from 'graphql-tag'
+import { useQuery } from '@apollo/react-hooks'
+import { BigNumber } from 'ethers/utils'
 
 interface Props {
   marketMakerAddress: string
 }
 
+const GET_COLLATERAL_VOLUME_NOW = gql`
+  query Current($id: String) {
+    fixedProductMarketMakers(where: { id: $id }) {
+      collateralVolume
+    }
+  }
+`
+
+const buildQuery24hsEarlier = (hash: Maybe<string>) => {
+  return gql`
+  query AfterHash($id: String) {
+    fixedProductMarketMakers(where: { id: $id }, block: { hash: "${hash}" }) {
+      collateralVolume
+    }
+  }
+`
+}
+
 const MarketViewContainer: React.FC<Props> = (props: Props) => {
   const context = useConnectedWeb3Context()
 
   const { marketMakerAddress } = props
 
+  const [hash, setHash] = useState<Maybe<String>>(null)
   const { marketMakerData, status } = useMarketMakerData(marketMakerAddress, context)
+  const { library: provider } = context
+
+  const [lastDayVolume, setLastDayVolume] = useState<Maybe<string>>(null)
+
+  const { data: volumeNow, error: errorVolumeNow } = useQuery(GET_COLLATERAL_VOLUME_NOW, {
+    skip: !!lastDayVolume,
+    variables: { id: marketMakerAddress.toLowerCase() },
+  })
+
+  const { data: volumeBefore, error: errorVolumeBefore } = useQuery(buildQuery24hsEarlier(hash && hash.toLowerCase()), {
+    skip: !!lastDayVolume,
+    variables: { id: marketMakerAddress.toLowerCase() },
+  })
+
+  if (errorVolumeBefore || errorVolumeNow) {
+    setLastDayVolume('-')
+  } else if (volumeNow && volumeBefore) {
+    const now = new BigNumber(volumeNow.fixedProductMarketMakers[0].collateralVolume)
+    const before = new BigNumber(volumeBefore.fixedProductMarketMakers[0].collateralVolume)
+
+    setLastDayVolume(now.sub(before).toString())
```

Comment:
I think `lastDayVolume` should be a BigNumber. Converting it to string is a view responsibility.